preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -118,10 +118,16 @@ def parse_nextclade_json(
     result_dir,
     nextclade_metadata: defaultdict[AccessionVersion, defaultdict[SegmentName, dict[str, Any]]],
     segment,
+    unaligned_nucleotide_sequences,
 ) -> defaultdict[AccessionVersion, defaultdict[SegmentName, dict[str, Any]]]:
     """
-    Update nextclade_metadata object with the results of the nextclade analysis
+    Update nextclade_metadata object with the results of the nextclade analysis.
+    If the segment existed in the input (unaligned_nucleotide_sequences) but did not align
+    nextclade_metadata[segment]=None.
     """
+    for id, seg_dict in unaligned_nucleotide_sequences.items():
+        if segment in seg_dict and seg_dict[segment] is not None:
+            nextclade_metadata[id][segment] = None
```

Comment:
```suggestion
    for id, segment_sequences in unaligned_nucleotide_sequences.items():
        if segment in segment_sequences and segment_sequences[segment] is not None:
            nextclade_metadata[id][segment] = None
```
If this dict is indeed like { "S": "ATTTTGG"...}